### ISSUE TITLE
unable to render valid html string

### ISSUE THREAD

**Issue by beckend** [Baptistemontan/leptos_i18n#11]
this one fails:

```html
<article><h2>Effortless Process: From Virtual Introduction to Contract with Our Talent Partner</h2><p>Our talent partner simplifies your recruitment journey for maximum efficiency, whether you're seeking candidates for hybrid, remote, or onsite positions. Our adaptable approach ensures you're informed every step of the way, encompassing permanent positions, headhunting, and contract limited time roles. Here's a detailed look at our process, including virtual meetings, contract signing, fee details, and the one-time replacement guarantee:<h3>1. Schedule Virtual Meeting Invite</h3><p>To kickstart our collaboration, schedule a virtual meeting invite. During this meeting, we'll discuss your recruitment needs, spanning hybrid, remote, or onsite positions, including permanent roles, headhunting, and contract limited time opportunities. This initial step sets the stage for our partnership.<h3>2. Understanding Our Versatile Approach</h3><p>Our recruitment model encompasses various options to suit your hiring requirements:<ul><li><strong>Contingency Hiring:</strong> This model is based on contingency, meaning we only generate revenue upon successfully placing an employee with you. Until that point, we don't charge any fees. The fees charged are influenced by various factors, including the complexity of the role, candidate's first-year salary, and adherence to relevant regulations. On average, typical recruitment fees range from 15% to 25% of the candidate's first-year salary.<li><strong>Retainer Model:</strong> Alternatively, we accept retainer model agreements. With this option, companies pay an upfront fee to create an exclusive contract for the recruitment process. This approach allows for greater dedication and personalized attention to your needs. Part of the fee is paid upfront, demonstrating commitment and enabling us to initiate the process promptly. The remaining payment is made at varying points of the process. Despite the variation, the total percentage paid remains within the range of 15% to 25% of the candidate's first-year salary.<li><strong>Contract Placement:</strong> For contract placements, we provide an internal payroll solution. We handle the employee's payroll and charge the client based on a comprehensive fee structure. This fee includes the employee's wages along with an additional amount to cover expenses. Typically, this fee equates to around 1.5 times the employee's wages. For instance, if an employee earns $20 per hour, the agency would charge the client $20 times 1.5, resulting in a rate of $30 per hour. Contract placements often pave the way for potential temp-to-Hire roles.</ul><p><strong>Guarantee and Refund Policy:</strong> Our one-time replacement guarantee ensures your satisfaction. If a candidate doesn't meet your expectations within the agreed period, we offer a replacement candidate. The guarantee covers a single replacement, and any additional terms will be outlined in the agreement. This policy is open to negotiation, ensuring that your unique needs are met.<p>Following our detailed discussion, we'll tailor agreements that encapsulate the fee structure, guarantee terms, and the chosen recruitment model, including our negotiable guarantee and refund policy. Rest assured, our process ensures compliance with regulations and dedication to your unique recruitment needs.<h3>3. Diligent Search</h3><p>Our search commences, leveraging our extensive network and expertise to identify ideal candidates.<h3>4. Candidate-Employer Match and Multiple Choices</h3><p>We meticulously evaluate candidates to ensure a seamless alignment with your criteria and company culture. In situations where you present multiple candidates, our evaluation extends to all choices.<h3>5. Candidate Interview Process and Contract Preparation</h3><p>As candidates are chosen continuously, this step continues until you find a candidate you like. We facilitate the interview scheduling process with your team. Our goal is to ensure seamless communication and coordination between the candidates and your employees, culminating in a successful mutual agreement. Following the successful interviews, comprehensive contracts are prepared for both parties involved.<h3>6. Collaboration for Agreement</h3><p>Contracts are shared for review, emphasizing open lines of communication. Once both the candidate and your team agree, the contracts are officially executed, solidifying the agreement.<h3>7. Payment After Placement - Invoice Sent Upon Start</h3><p>Following the candidate's start date, an invoice will be promptly sent to your organization for payment. This invoice corresponds to the agreed-upon fee structure as outlined in the agreement. The payment is due within 14 days from the candidate's start date.<h3>9. Continuing Partnership</h3><p>Our dedication endures past placement. We ensure a smooth transition and nurture lasting relationships.<p>Our process ensures you're informed, from scheduling a virtual meeting invite to candidate placement across various positions. The adaptable fee structure, aligned with your chosen recruitment model, intrinsically tied to the complexity of the role and the candidate's first-year salary, reflects our dedication to streamlining your hiring process.<p>Experience recruitment ease with our talent partner. With the added assurance of our one-time replacement guarantee and compliance with relevant regulations, we're steadfastly with you at every step.</article>
```

```shell
error[E0599]: no method named `comp_h3` found for struct `__page_home_section_process_content_builder` in the current scope
  --> src/components/home/section_process.rs:21:10
   |
16 |         {i18n::t!(
   |  ________-
17 | |         i18n,
18 | |         page_home_section_process_content,
19 | |         <article> = |cx, children| view!{ cx, <article class=format!("col-xs-12 {}", ClassName::CONTENT)>{chil...
20 | |         <h2> = |cx, children| view!{ cx, <h2>{children(cx)}</h2> },
21 | |         <h3> = |cx, children| view!{ cx, <h3>{children(cx)}</h3> },
   | |         -^^ help: there is a method with a similar name: `comp_h2
```
It's minified and verified valid HTML.

The rust code used:
```rust
     {i18n::t!(
        i18n,
        page_home_section_process_content,
        <article> = |cx, children| view!{ cx, <article class=format!("col-xs-12 {}", ClassName::CONTENT)>{children(cx)}</article> },
        <h2> = |cx, children| view!{ cx, <h2>{children(cx)}</h2> },
        <h3> = |cx, children| view!{ cx, <h3>{children(cx)}</h3> },
        <p> = |cx, children| view!{ cx, <p>{children(cx)}</p> },
        <strong> = |cx, children| view!{ cx, <strong>{children(cx)}</strong> },
        <ul> = |cx, children| view!{ cx, <ul>{children(cx)}</ul> },
        <li> = |cx, children| view!{ cx, <li>{children(cx)}</li> }
      )}
```

**Comment by beckend** [comment]
The same code using one line unminified HTML works:

```html
<article>\\n  <h2>Effortless Process: From Virtual Introduction to Contract with Our Talent\\n    Partner</h2>\\n  <p>We simplifies your recruitment journey for maximum\\n    efficiency, whether you're seeking candidates for hybrid, remote, or onsite\\n    positions. Our adaptable approach ensures you're informed every step of the\\n    way, encompassing permanent positions, headhunting, and contract limited\\n    time roles. Here's a detailed look at our process, including virtual\\n    meetings, contract signing, fee details, and the one-time replacement\\n    guarantee:</p>\\n\\n  <h3>1. Schedule Virtual Meeting Invite</h3>\\n  <p>To kickstart our collaboration, schedule a virtual meeting invite. During\\n    this meeting, we'll discuss your recruitment needs, spanning hybrid, remote,\\n    or onsite positions, including permanent roles, headhunting, and contract\\n    limited time opportunities. This initial step sets the stage for our\\n    partnership.</p>\\n\\n  <h3>2. Understanding Our Versatile Approach</h3>\\n  <p>Our recruitment model encompasses various options to suit your hiring\\n    requirements:</p>\\n  <ul>\\n    <li><strong>Contingency Hiring:</strong> This model is based on contingency,\\n      meaning we only generate revenue upon successfully placing an employee\\n      with you. Until that point, we don't charge any fees. The fees charged are\\n      influenced by various factors, including the complexity of the role,\\n      candidate's first-year salary, and adherence to relevant regulations. On\\n      average, typical recruitment fees range from 15% to 25% of the candidate's\\n      first-year salary.</li>\\n    <li><strong>Retainer Model:</strong> Alternatively, we accept retainer model\\n      agreements. With this option, companies pay an upfront fee to create an\\n      exclusive contract for the recruitment process. This approach allows for\\n      greater dedication and personalized attention to your needs. Part of the\\n      fee is paid upfront, demonstrating commitment and enabling us to initiate\\n      the process promptly. The remaining payment is made at varying points of\\n      the process. Despite the variation, the total percentage paid remains\\n      within the range of 15% to 25% of the candidate's first-year salary.</li>\\n    <li><strong>Contract Placement:</strong> For contract placements, we provide\\n      an internal payroll solution. We handle the employee's payroll and charge\\n      the client based on a comprehensive fee structure. This fee includes the\\n      employee's wages along with an additional amount to cover expenses.\\n      Typically, this fee equates to around 1.5 times the employee's wages. For\\n      instance, if an employee earns $20 per hour, the agency would charge the\\n      client $20 times 1.5, resulting in a rate of $30 per hour. Contract\\n      placements often pave the way for potential temp-to-Hire roles.</li>\\n  </ul>\\n  <p><strong>Guarantee and Refund Policy:</strong> Our one-time replacement\\n    guarantee ensures your satisfaction. If a candidate doesn't meet your\\n    expectations within the agreed period, we offer a replacement candidate. The\\n    guarantee covers a single replacement, and any additional terms will be\\n    outlined in the agreement. This policy is open to negotiation, ensuring that\\n    your unique needs are met.</p>\\n  <p>Following our detailed discussion, we'll tailor agreements that encapsulate\\n    the fee structure, guarantee terms, and the chosen recruitment model,\\n    including our negotiable guarantee and refund policy. Rest assured, our\\n    process ensures compliance with regulations and dedication to your unique\\n    recruitment needs.</p>\\n\\n  <h3>3. Diligent Search</h3>\\n  <p>Our search commences, leveraging our extensive network and expertise to\\n    identify ideal candidates.</p>\\n\\n  <h3>4. Candidate-Employer Match and Multiple Choices</h3>\\n  <p>We meticulously evaluate candidates to ensure a seamless alignment with\\n    your criteria and company culture. In situations where you present multiple\\n    candidates, our evaluation extends to all choices.</p>\\n\\n  <h3>5. Candidate Interview Process and Contract Preparation</h3>\\n  <p>As candidates are chosen continuously, this step continues until you find a\\n    candidate you like. We facilitate the interview scheduling process with your\\n    team. Our goal is to ensure seamless communication and coordination between\\n    the candidates and your employees, culminating in a successful mutual\\n    agreement. Following the successful interviews, comprehensive contracts are\\n    prepared for both parties involved.</p>\\n\\n  <h3>6. Collaboration for Agreement</h3>\\n  <p>Contracts are shared for review, emphasizing open lines of communication.\\n    Once both the candidate and your team agree, the contracts are officially\\n    executed, solidifying the agreement.</p>\\n\\n  <h3>7. Payment After Placement - Invoice Sent Upon Start</h3>\\n  <p>Following the candidate's start date, an invoice will be promptly sent to\\n    your organization for payment. This invoice corresponds to the agreed-upon\\n    fee structure as outlined in the agreement. The payment is due within 14\\n    days from the candidate's start date.</p>\\n\\n  <h3>9. Continuing Partnership</h3>\\n  <p>Our dedication endures past placement. We ensure a smooth transition and\\n    nurture lasting relationships.</p>\\n\\n  <p>Our process ensures you're informed, from scheduling a virtual meeting\\n    invite to candidate placement across various positions. The adaptable fee\\n    structure, aligned with your chosen recruitment model, intrinsically tied to\\n    the complexity of the role and the candidate's first-year salary, reflects\\n    our dedication to streamlining your hiring process.</p>\\n\\n  <p>Experience recruitment ease with our talent partner. With the added\\n    assurance of our one-time replacement guarantee and compliance with relevant\\n    regulations, we're steadfastly with you at every step.</p>\\n</article>\\n
```

**Comment by Baptistemontan** [comment]
Hello, the current algorithm to parse the html strings is a bit dumb, it basically looks for a `<tag>` then try to find a corresponding `</tag>`, taking into account the new `<tag>` in the way. 

The problem here is that in the first html string there is no `</p>`, so it skips it. If the code don't find a valid closing tag it silently abort. I need to change this behavior, the second one works because the closing tags `</p>` are present.

I don't really know if I want to support non closing `p` elements, it's a lot of work for a special edge cases that should not be needed in normal uses, this feature was'nt really meant to plug full blown pages into it.

**Comment by beckend** [comment]
That makes sense thanks

**Comment by Baptistemontan** [comment]
Thanks for Issue, the code currently don't error when a `<tag>` is present but no matching closing tags are present, because I don't know how to reject false positive yet (I got some ideas but did not had time to implement them), but this will probably be a hard error in the future.

**Comment by Baptistemontan** [comment]
I fixed a parsing bug in #13 where the parsing aborted when no closing tag was found, but the expected behavior for `<p>` with no closing tag is the same, it is silently ignored. so in the minified version it should give an error when trying to supply the `<p>` in the `t!` macro, but not the others tags.

I'm happy where it is now, no hard errors if no closing tag is found, because false positive on opening tag could easily happend and appart from introducing weird syntax to deny them, I can't think of a good way to prevent it.

I'm closing this issue as it comes from the minified version of p elements, and I don't plan on supporting it right now, maybe in the future.

I would be glad to have any feedback, feel free to open issues for any problems.
